### PR TITLE
Updates Travis Trusty base image

### LINKED ISSUE #1876: Update Travis CI to new Trusty image
# Feature request

## Description

We're currently using the deprecated Ubuntu Trusty image in the Travis CI run to sidestep problems with the [updated version](https://blog.travis-ci.com/2017-06-21-trusty-updates-2017-Q2-launch). The deprecated image will be retired on 2017-09-01, so we need to resolve the functional test issues prior to that date. 

See #1874 for details. #1655 may aid in resolution.

## User Stories

As a SecureDrop developer, I want the application test suite to run in CI beyond 2017-09-01, so that I can be confident the changes I make to the application code do not introduce bugs or regressions.

### PR DESCRIPTION
**Pull request by conorsch**
## Status

Ready for review.

## Description of Changes

Fixes #1876.

Changes proposed in this pull request:

* Most importantly, uses the new Travis Trusty image, since the currently used one expires on 2017-09-01.
* Disables the page-layout tests that came in from #2141. Will need to reactivate them in a separate PR.
* Uses the Travis-provided version of Firefox 46.0.1 for Selenium
* Dumps Selenium log output automatically in Travis runs
* Reorganizes (but doesn't much change) the Firefox install tasks
* Beefs up the config tests regarding what version of Firefox is considered proper
* Contains some Travis-specific customizations to the test runs

The final item, the Travis-specific changes in the test suites, I strongly dislike, but these tests are undergoing some major changes in the near future, so I'm fine with it given the breakage deadline of 2017-09-01.

## Testing
* [x] Confirm all CI passes!
* [x] Confirm the following in a dev VM:
  * [x] app tests pass
  * [x] page layout tests pass
  * [x] config tests pass

If all those conditions are met, I'm satisfied for merge.

## Deployment

No, these are development- and CI-specific changes.

### If you made changes to the app code:

- [x] Unit and functional tests pass on the development VM

### If you made changes to the system configuration:

- [x] [Configuration tests](https://docs.securedrop.org/en/latest/development/testing_configuration_tests.html) pass

### If you made changes to documentation:

- [ ] Doc linting passed locally

### CODE REVIEW

**Comment by conorsch** [comment]
Travis tests are failing. :rage1: Perhaps I screwed something up while rebasing. Will confirm passing tests on personal fork again, then update this branch. Marked as WIP.

**Comment by conorsch** [comment]
Ah-ha! #2215 got merged just before this, and I fumbled the rebase. Will try again in light of those changes.

**Comment by conorsch** [comment]
Rebased! Confirmed working build of new code in personal fork: https://travis-ci.org/conorsch/securedrop/builds/270131377 Now we wait for CI to pass here. 🎲

**Comment by conorsch** [comment]
Travis is passing:

travis-works-again

However CircleCI failed due to transient AWS problems. Restarting....

**Comment by dachary** [comment]
Awesome work @conorsch. You gained 5 points in stamina against rogue CI services.

**Comment by conorsch** [comment]
@dachary 😅 

All CI is passing. Marked as "Ready for review."

File: install_files/ansible-base/roles/app-test/tasks/setup_firefox_for_selenium.yml [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    - firefox_installed_result.stdout != "Mozilla Firefox 46.0.1"
+
+# Selenium 3 makes breaking changes with the 2.X API, and requires the
+# installation of the Mozilla geckodriver. Since the Aaron Swartz Day Hackathon
+# is approaching, which will involve many new external contributors, we've
+# decided to play it as safe as possible by downgrading Firefox to the latest
+# version (46.0.1) that is compatible with the last 2.X series Selenium release
+# (2.53.6). After the Hackathon, we'll resolve the geckodriver business and
+# remove the following three tasks (as well as add firefox back to the
+# `test_apt_dependencies` list).
```
Comment (redshiftzero) [inline install_files/ansible-base/roles/app-test/tasks/setup_firefox_for_selenium.yml]: 😬

File: securedrop/tests/functional/functional_test.py [hunk 2]
```diff
@@ -78,7 +78,17 @@ def _prepare_webdriver(self):
...
+            # In Travis, the Firefox path isn't in /usr/bin, and Selenium
+            # doesn't find it automatically.
+            firefox_path = "/home/travis/firefox-46.0.1/firefox/firefox"
+            firefox = firefox_binary.FirefoxBinary(firefox_path,
+                                                   log_file=log_file)
+        else:
+            # Selenium's PATH inference is smart enough outside Travis.
+            firefox = firefox_binary.FirefoxBinary(log_file=log_file)
+
+        return firefox
```
Comment (redshiftzero) [inline securedrop/tests/functional/functional_test.py]: I also don't love this Travis-specific code path, but I understand why you did this and will not futz. I'll file a followup issue though so on a rainy day in the future someone can workaround this (e.g. by adding the path to the Firefox binary to the PATH in the Travis config).

Comment (conorsch) [reply securedrop/tests/functional/functional_test.py]: > e.g. by adding the path to the Firefox binary to the PATH in the Travis config

Perplexingly that may not work: the Travis-supplied Firefox binary is _already_ in PATH for the travis user! Selenium uses `which` to find the binary, and I _suspect_ that executing the tasks as root breaks the PATH logic. Will find out while digging through #2216.

A discrete issue is well warranted.

**Pull request merged**